src/utils/renderNamePattern.ts

```diff
@@ -31,12 +31,15 @@ export function renderNamePattern(
 }
 
 export function makeUnique(name: string, existingNames: Record<string, any>) {
-	if (name in existingNames === false) return name;
-
-	const ext = extname(name);
-	name = name.substr(0, name.length - ext.length);
-	let uniqueName,
-		uniqueIndex = 1;
-	while (existingNames[(uniqueName = name + ++uniqueIndex + ext)]);
-	return uniqueName;
+	for (const existingName of Object.keys(existingNames)) {
+		if (name.toLowerCase() === existingName.toLowerCase()) {
+			const ext = extname(name);
+			name = name.substr(0, name.length - ext.length);
+			let uniqueName,
+				uniqueIndex = 1;
+			while (existingNames[(uniqueName = name + ++uniqueIndex + ext)]);
```

Comment:
If there are files `Date.js` and `Date1.js`, a file named `date.js` would be renamed `date1.js` and thus conflict with the other name because you are not being case-insensitive here. For a quick fix, instead of iterating over the existing names you could create a Set a-priori to reference against:

```js
const existingNamesLowercase = new Set(Object.keys(existingNames).map(key => key.toLowerCase()));
if (!existingNamesLowercase.has(name) return name;
...
while (existingNamesLowercase.has((uniqueName = name + ++uniqueIndex + ext).toLowerCase());
```